<html><body><div><div id="disqus_post_message"><p>Сегодня ВНЕЗАПНО все <a href="http://www.celeryproject.org/">Celery</a> воркеры перестали принимать любые задачи, при том что <a href="http://www.rabbitmq.com/">RabbitMQ</a> (используется в качестве брокера) очереди были пустыми, а остальные части системы функционировали нормально. Не помогала ни перезагрузка Celery воркеров, ни перезагрузка RabbitMQ сервера.</p>

<p>После недолгого копания по логам проблема была локализирована, любой Celery воркер как будто зависал на моменте поиска соседей, оставляя в логах что-то такое:</p>

<pre><code>2014-08-25 14:29:12 [INFO:celery.worker.consumer] Connected to amqp://guest:**@127.0.0.1:5672//
2014-08-25 14:29:12 [INFO:celery.worker.consumer] mingle: searching for neighbors
</code></pre>

<p>Быстрый гуглинг указал на <a href="https://github.com/celery/celery/issues/1664" title="mingle: searching for neighbors">существующую Celery issue</a>, а уже в ней и нашелся ответ на проблему. Оказывается на корневом разделе подошло к концу место (было доступно порядка 150 Мб) и Celery в связке с RabbitMQ зависала не показывая никаких признаков жизни. Все починилось банальной чисткой корневого раздела, но осадок остался и на себя, что не поставил уведомления о заканчивающемся месте на корневом разделе и на Celery, что она никаким образом не пытается обработать эту ситуацию и проблему приходится вычислять окольными путями.</p>
</div>


</div></body></html>